addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -57,7 +57,12 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
         }
 
         if (result.status === true) {
-            $amountDelivery.text(result.new_amount_delivery);
+            // if free delivery (`free_over` field), show 'Free', not '$0'
+            if (result.new_amount_delivery) {
+                $carrierBadge.children('span').text(result.new_amount_delivery);
+            } else {
+                $carrierBadge.text(_t('Free'));
+            }
```

Comment:
We only go through here once ?

Reply:
Without my improvement, so with current code, we go through this method every time you click on a carrier.
With my improvement, so still go in here on every carrier click + once per carrier at widget start since it fires a RPC for every carrier that callback this method.

It is just an small improvement to show 'Free' and not '$0' as currently free delivery thanks to coupon show 'Free' but free delivery thanks to delivery module (free over field) show '$0'.

Reply:
If we go through this multiple times, if we go through the `else` part, you will remove the "span" that the `if` needs..?

Reply:
Oh, I did not get what you meant. No this is not a possible flow.
We might go through this code as many times as the client clicks on the carrier but the delivery is not supposed to become suddently free or suddently not free, we will go through the same condition every times.

(So yes the `span` will be gone but we won't ever go in the if condition. It is a bit weird but it is how amount is updated, as it is a currency widget it is composed of 2 `span`, one for the currency symbol and one for the amount, every time the RPC brings back an amount, only the amount `span` is updated.
That's why I replace the badge and not the span (as coupon already does before this PR).)